### PR TITLE
Clean up the scholarship data handler to fix linter and compiler warnings

### PR DESCRIPTION
**Pull request by CarlosM10011**
This pr fixes some compiler warnings in the scholarship data handler.

### CODE REVIEW

**Review by MikolJiron — APPROVED** [review]

**Comment by Lorraine-Lyu** [comment]
@googlebot I consent.

**Review by Lorraine-Lyu — APPROVED** [review]

File: src/main/webapp/js/datahandlers/scholarshipdatahandler.js [hunk 1]
```diff
@@ -19,33 +19,35 @@ const {Map: SoyMap} = goog.require('soy.map');
...
     const requirementsMap = new Map();
-                           
+
     let requirement = undefined;
     for (requirement of REQUIREMENTS) {
-      if (REQUIREMENTS[requirement] != undefined) {
-        requirementsMap.set(addSpaceToCamelCase(requirement), REQUIREMENTS[requirement].join(SEPARATOR));
+      if (data[requirement] != undefined) {
+        requirementsMap.set(
+            addSpaceToCamelCase(requirement),
```
Comment (tinagrasso) [inline src/main/webapp/js/datahandlers/scholarshipdatahandler.js:49]: Why not just map the field to the display name?

File: src/main/webapp/js/datahandlers/scholarshipdatahandler.js [hunk 2]
```diff
@@ -19,33 +19,35 @@ const {Map: SoyMap} = goog.require('soy.map');
...
+            data[requirement].join(SEPARATOR));
       } else {
         requirementsMap.set(addSpaceToCamelCase(requirement), NA);
```
Comment (tinagrasso) [inline src/main/webapp/js/datahandlers/scholarshipdatahandler.js:52]: nit: don't substitute empty values. It clogs up your UI and pollutes the valuable text. I would leave them blank or just use a dash if you really want something there.

**Review by tinagrasso — CHANGES_REQUESTED** [review]

File: src/main/webapp/js/datahandlers/scholarshipdatahandler.js (hunk 1 above)
Comment (hzimz) [reply src/main/webapp/js/datahandlers/scholarshipdatahandler.js:49]: +1

**Review by hzimz — APPROVED** [review]

**Comment by CarlosM10011** [comment]
So I'm working on an older version of the file; going to close this because it looks like many of these changes may get rewritten.

**Pull request closed without merging**